src/pages/workspaces/workspace/WorkspaceTabs.ts

```diff
@@ -3,9 +3,9 @@ import { Dispatch, ReactNode } from 'react';
 import { h } from 'react-hyperscript-helpers';
 import { analysisTabName } from 'src/analysis/runtime-common-components';
 import { TabBar } from 'src/components/tabBars';
+import { isProtectedWorkspace } from 'src/import-data/protected-data-utils';
 import * as Nav from 'src/libs/nav';
 import {
-  hasProtectedData,
```

Comment:
Does this mean that `hasProtectedData` should be removed? Or is the idea that `hasProtectedData` is Azure-specific, but `isProtectedWorkspace` is not?

Reply:
[isProtectedWorkspace ](https://github.com/DataBiosphere/terra-ui/blob/dev/src/import-data/protected-data-utils.ts#L96 )calls hasProtectedData if the workspace is an Azure one, so just upleveling the logic to support both types of clouds.

Reply:
Has UX/product confirmed that they do want the icon showing for a GCP workspace with an auth domain? That wasn't the original intent, but perhaps it has changed (in which case I will approve the PR).

![image](https://github.com/DataBiosphere/terra-ui/assets/484484/b643fee9-030a-4717-9f3a-f3137cc617e2)

Reply:
Ugh, I just read https://github.com/DataBiosphere/terra-ui/pull/4479#issuecomment-1816985104. Could you also add a unit test case for a GCP workspace with an auth domain showing this icon? And I don't know how you are handling the "additional security logging" checkbox, but if there's a way to add a test case for that as well it would be good.

Reply:
I am trusting that the current logic for determining if a workspace is protected or not covers these cases. There are tests for that function here for both Azure and GCP: https://github.com/DataBiosphere/terra-ui/blob/b31e3cb5269f253201df267c5879620cf8aa618e/src/import-data/protected-data-utils.test.ts#L95C1-L95C1

there are also tests for the shield that you added, which basically will ensure that if the protected message is there the shield will show up (the isProtectedWorkspace call will decide that) . 

For the new test, would it be combining the two tests to work together? it feels like the coverage is there but it is not joined together but in my mind it is sufficient but I can be convinced otherwise 😄

Reply:
actually I see what you mean, ignore my previous comment!

Reply:
I am getting to the point of accepting that if `isProtectedWorkspace` has good coverage, we don't also need it here.

But I'm surprised to see that auth domains weren't checked in the test for `isProtectedWorkspace`. Perhaps that's because the bucket name somehow implies there is an auth domain?

Reply:
I also noticed that and was wondering. I'll check with Beth and create a follow up ticket for this if further work is needed! 

I ended up adding a test and actually found a place where hasProtectedData instead of isProtectedWorkspace so that was a good call 😺